segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/operations/MergeSegmentOperation.java

```diff
@@ -47,6 +54,12 @@ public MergeSegmentOperation(long targetSegmentId, long sourceSegmentId) {
         this.streamSegmentOffset = -1;
     }
 
+    public MergeSegmentOperation(long targetSegmentId, long sourceSegmentId, AttributeUpdateCollection attributeUpdates) {
+        this(targetSegmentId, sourceSegmentId);
+        // this.attributeUpdates is already set as an empty collection in the above constructor. Overwrite it only if needed.
+        this.attributeUpdates = attributeUpdates == null ? this.attributeUpdates : attributeUpdates;
```

Comment:
I don't understand what exactly is in `this.attributes` (the first part of the `?` operator)? It's null at this point. 
I think just setting `this.attributeUpdates = attributeUpdates` should suffice.

EDIT: I see you initialize it to an empty list. Then it makes sense. However I would prefer if the behavior is consistent with `UpdateAttributesOperation` and `StreamSegmentAppendOperation`.  Those do allow `null` for this.

Reply:
Done, now it can accept `null` values. But note that `UpdateAttributesOperation` actually does not accept `null` values in `attribtueUpdates`, as it has a `Preconditions.checkNotNull` in its constructor.